gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/GremlinDriverIntegrateTest.java

```diff
@@ -1879,4 +1879,98 @@ public void shouldNotHangWhenSameRequestIdIsUsed() throws Exception {
 
         assertEquals(100, result1.get().one().getInt());
     }
+
+    @Test
+    public void shouldFailOnInitiallyDeadHost() throws Exception {
+        // Stop server
+        this.stopServer();
+
+        final Cluster cluster = TestClientFactory.build().create();
+        final Client client = cluster.connect();
+
+        try {
+            // try to re-issue a request now that the server is down
+            System.out.println("SUBMIT 1");
+            client.submit("g").all().get(3000, TimeUnit.MILLISECONDS);
+            fail("Should throw an exception.");
+        } catch (RuntimeException re) {
+            // Client would have no active connections to the host, hence it would encounter a timeout
+            // trying to find an alive connection to the host.
+            Throwable foo = re.getCause();
+            assertThat(re.getCause(), instanceOf(NoHostAvailableException.class));
+
+            //
+            // should recover when the server comes back
+            //
+
+            // restart server
+            this.startServer();
+
+            // try a bunch of times to reconnect. on slower systems this may simply take longer...looking at you travis
+            for (int ix = 1; ix < 11; ix++) {
+                // the retry interval is 1 second, wait a bit longer
+                TimeUnit.SECONDS.sleep(5);
+
+                try {
+                    System.out.println("SUBMIT " + (1 + ix));
+                    final List<Result> results = client.submit("1+1").all().get(3000, TimeUnit.MILLISECONDS);
```

Comment:
Again would suggest a timeout of like 500 ms. No need to have a huge timeout when it is a local connection, it will either reply nearly instantly or not at all.